src/loaders/RasterJsonLoader.js

```diff
@@ -12,34 +13,60 @@ async function initLoader(imageData) {
   } = imageData;
   switch (type) {
     case ('zarr'): {
-      const { dimensions, isPyramid, transform } = metadata || {};
-      const { scale = 0, translate = { x: 0, y: 0 } } = transform;
-      const loader = await createZarrLoader({
-        url, dimensions, isPyramid, scale, translate,
-      });
-      return loader;
+      const {
+        dimensions, isPyramid, transform,
+      } = metadata || {};
+      const labels = dimensions.map(d => d.field);
+      let source;
+      if (isPyramid) {
+        const metadataUrl = `${url}${
+          url.slice(-1) === '/' ? '' : '/'
+        }.zmetadata`;
```

Comment:
Ah, `createZarrLoader` created a loader for a legacy layout for arr pyramids. Maybe we can extract this whole thing into a `createLegacyZarrSource`.